src/st/create_entities_horizontal.h

```diff
@@ -21,17 +21,17 @@ void CreateEntitiesToTheLeft(s16 posX) {
     }
 
     while (true) {
-        if (g_LayoutObjHorizontal->posX == LAYOUT_OBJ_START ||
-            posX > g_LayoutObjHorizontal->posX) {
+        if (g_LayoutObjHorizontal[0] == LAYOUT_OBJ_START ||
```

Comment:
`g_LayoutObjHorizontal[LAYOUT_OBJ_POS_X]` etc?

Reply:
It wouldn't make sense in this context because the first element wouldn't indicate Position X but just the start of the data.

Same for `InitRoomEntities` ([scratch](https://decomp.me/scratch/thtl9), where the array elements `0` and `1` from `g_LayoutObjHorizontal = (u16*)((g_LayoutObjHorizontal[1] << 0x10) + g_LayoutObjHorizontal[0])` do not indicate PosX or PosY but just two halves or a pointer array... it is unfortunately horrible, hacky and very not-well structured.